src/external-server/docker-amarna.ts

```diff
@@ -28,16 +27,29 @@ export class AmarnaDocker extends DockerServer {
                 );
             }
         }
-        return dockerArgs;
+        return cmd;
     }
 
     public async run(args: { script?: boolean }) {
+        const { path } = this;
+        if (!this.docker) {
+            this.docker = await HardhatDocker.create();
+        }
         this.useShell = !!args.script;
-        console.log("Running amarna, this may take a while.");
-        await this.spawnChildProcess();
-    }
+        const formattedImage = `${this.image.repository}:${this.image.tag}`;
+        if (!this.docker.hasPulledImage(this.image)) {
+            console.log(`Pulling amarna image ${formattedImage}.`);
+            await this.docker.pullImage(this.image);
+        }
+        const cmd = this.getDockerArgs();
 
-    protected async getContainerArgs(): Promise<string[]> {
-        return [];
+        console.log("Running amarna, this may take a while.");
+        const { stdout, stderr } = await this.docker.runContainer(this.image, cmd, {
+            binds: {
+                [path]: "/src"
```

Comment:
Why doesn't it need to bind other paths as well? Like source paths?

Reply:
Resolving this conversation, continued in https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/302#discussion_r1090316707